lib/queue.rb

```diff
@@ -1,31 +1,49 @@
 class Queue
-
+  
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = -1
+    @back = -1
   end
-
+  
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
-  end
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 0
+    end
+
+    if @front == @back
+      #decide
+    end
```

Comment:
This part works better as an elsif to raise an error if the queue is full.  You're not doing anything here.